tools/docs/sphinx_runner.py

```diff
@@ -0,0 +1,202 @@
+import argparse
+import os
+import platform
+import re
+import sys
+import tarfile
+import tempfile
+from functools import cached_property
+
+import yaml
+
+from colorama import Fore, Style
+
+from sphinx.cmd.build import main as sphinx_build
+
+from tools.base import runner
+
+
+class SphinxBuildError(Exception):
+    pass
+
+
+class SphinxEnvError(Exception):
+    pass
+
+
+class SphinxRunner(runner.Runner):
+    _build_dir = "."
+    _build_sha = "UNKNOWN"
+
+    @property
+    def blob_sha(self) -> str:
+        return self.docs_tag if self.docs_tag else self.build_sha
+
+    @property
+    def build_dir(self) -> str:
+        return self._build_dir
+
+    @property
+    def build_sha(self) -> str:
+        return self.args.build_sha or self._build_sha
+
+    @cached_property
+    def colors(self) -> dict:
+        return dict(chrome=Fore.LIGHTYELLOW_EX, key=Fore.LIGHTCYAN_EX, value=Fore.LIGHTMAGENTA_EX)
+
+    @cached_property
+    def config_file(self) -> str:
+        with open(self.config_file_path, "w") as f:
+            f.write(yaml.dump(self.configs))
```

Comment:
@dmitri-d in this case it *does* write to the config file - and here i think this is ok - we only want it to happen once, and on first access to this property

its arguable that it would be better being explicit because of it persisting, but tbh its still ephemeral and im not sure i see the benefit

Reply:
Mostly readability/maintenance issue in this case; I certainly wouldn't expect such a behaviour from an accessor. If you think the implementation is warranted, could you add a comment mentioning that there's a side-effect in this method that dumps the config to a file, or something along these lines...

Reply:
sure, ill update

Reply:
done